api/v1alpha1/backendtrafficpolicy_types.go

```diff
@@ -103,6 +104,15 @@ type BackendTrafficPolicySpec struct {
 	// +optional
 	// +notImplementedHide
 	Compression []*Compression `json:"compression,omitempty"`
+
+	// ConnectionBufferLimitBytes Soft limit on size of the cluster’s connections read and write buffers.
+	// If unspecified, an implementation defined default is applied (32768 bytes).
+	// For example, 20Mi, 1Gi, 256Ki etc.
+	// Note: that when the suffix is not provided, the value is interpreted as bytes.
+	//
+	// +kubebuilder:validation:XValidation:rule="type(self) == string ? self.matches(r\"^[1-9]+[0-9]*([EPTGMK]i|[EPTGMk])?$\") : type(self) == int",message="ConnectionBufferLimitBytes must be of the format \"^[1-9]+[0-9]*([EPTGMK]i|[EPTGMk])?$\""
+	// +optional
+	ConnectionBufferLimitBytes *resource.Quantity `json:"connectionBufferLimitBytes,omitempty"`
```

Comment:
should we move this under a `connection` field ? do we see any other values being added in here ?
```
connection:
  bufferLimit
```

cc @envoyproxy/gateway-maintainers

Reply:
```yaml
connection:
  bufferLimit
```
is better for first look.

Reply:
If we wanted to stay in a backwards compatible way we could set it under "connection", but at the moment I think it might be more appropriate to just use one field (_as it's not yet clear what additional fields will be added_). what do you think?

@arkodg  :)

Reply:
In the end, I chose to add a parent type `Connection` to it instead of just using a single field.
The reason can be seen: https://github.com/envoyproxy/gateway/pull/3390#issuecomment-2119790669